pkg/sbproxy/reconcile/reconciler.go

```diff
@@ -36,6 +39,43 @@ type Resyncer interface {
 	Resync(ctx context.Context)
 }
 
+// TimestamppedError contains an error and a timestamp in time.RFC3339Nano format
+type TimestamppedError struct {
+	Cause error
+}
+
+func (te TimestamppedError) Error() string {
+	return fmt.Sprintf("%s-%s", time.Now().Format(time.RFC3339Nano), te.Cause)
```

Comment:
This will be evaluated when the function is invoked. Therefore the timestamp would be the time when the error is printed and not when occurred. Probably misleading timestamp with the current implementation.